From 8d7b4f0fe0a0b904e33f53b50bc92d7705cd9390 Mon Sep 17 00:00:00 2001
From: Ćukasz Patron <priv.luk@gmail.com>
Date: Thu, 22 May 2025 20:55:57 +0200
Subject: [PATCH] type hints: typing.Optional[...] -> typing.Optional[typing.Union[...]]

Fixes the following error on Python 3.14:

TypeError: typing.Optional requires a single type. Got (<class
'PySide6.QtGui.QKeySequence'>, <class 'PySide6.QtCore.QKeyCombination'>,
<enum 'StandardKey'.

Change-Id: Id1fe45d31cc38a337caa55a86f509c3f88c0b683
---

diff --git a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/layout.py b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/layout.py
index 109562a..8eea431 100644
--- a/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/layout.py
+++ b/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/signature/layout.py
@@ -352,6 +352,7 @@
                 if last == _VAR_POSITIONAL:
                     kind = _KEYWORD_ONLY
         if default is None:
+            ann = typing.Union[ann]
             ann = typing.Optional[ann]
         if default is not _empty and layout.ellipsis:
             default = ellipsis
@@ -373,6 +374,7 @@
 
     ret_anno = annotations.get('return', _empty)
     if ret_anno is not _empty and props["fullname"] in missing_optional_return:
+        ret_anno = typing.Union[ret_anno]
         ret_anno = typing.Optional[ret_anno]
     return inspect.Signature(params, return_annotation=ret_anno,
                              __validate_parameters__=False)
